scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,54 +58,52 @@ def generate_load_table_files(dataset_tables_dict):
-    return dataset_tables_dict
-
-
-def validate_dataset(dataset):
-    """Validate the full dataset metadata with Schema validator."""
+def validate_dataset(dataset, schema_dict, field_dict):
+    validated_dataset = {}
+
+    for table in list(schema_dict.keys()):
+        if table not in list(dataset.keys()):
+            print(f"Failed: Missing required table {table}.")
+            print(dataset.keys())
+            exit()
+
+        errors = dynamically_validate_df(
+            field_dict=field_dict, 
+            fields_to_validate_list=schema_dict[table], 
+            data_df=dataset[table]
+            )
+        # get indices of the failed rows
+        errors_index_rows = [e.row for e in errors]
+
+        # if any errors, print error message and exit
+        if errors_index_rows:
+            print(f"Failed: Dataset validation errors found in table {table}. Please retry after correcting errors listed in validation_errors.csv.")
+            pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
+            exit()
+
+        # TODO: determine how to handle drops of failed rows/columns to get cleaned data
+        validated_dataset[table] = dataset[table]
+        print(f"Success: Table {table} has been validated.")
+    
+    print(f"Success: Dataset has been validated.")
+    return validated_dataset
 
-    # capture errors from the validation against the pre-defined validations per column
-    errors = schema_validator.validate(dataset, columns=schema_validator.get_column_names())
-    # get indices of the failed rows
-    errors_index_rows = [e.row for e in errors]
 
-    # if any errors, print error message and exit
-    if errors_index_rows:
-        print(f"Failed: Dataset validation errors found. Please retry after correcting errors listed in validation_errors.csv.")
-        pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
-        exit()
+def col_is_in_defined_fields(col_name, field_dict):
+    list_of_fields = list(field_dict.keys())
+    if col_name in list_of_fields:
+        return True
 
-    # TODO: determine how to handle drops of failed rows/columns to get cleaned data
-    # errors_index_rows = [e.row for e in errors]
-    # data_clean = dataset_metadata_df.drop(index=errors_index_rows, axis=1)
-    # data_clean.to_csv('clean_data.csv')
-    print(f"Success: Dataset has been validated.")
-    return dataset
+    return False
 
 
-def load_excel_input(excel, expected_dataset_cols):
+def load_excel_input(excel, allowed_dataset_cols):
     """Read excel file input into a dataframe and validate for required columns."""
 
     try:
-        dataset_metadata_df = pd.read_excel(excel, sheet_name="Sheet1", skiprows=2, usecols=expected_dataset_cols, index_col=None)
-
+        raw_dataset_df = pd.read_excel(excel, sheet_name=None, skiprows=2, index_col=None, usecols=lambda x: x in allowed_dataset_cols)
+        
+    # TODO: Add better file error handling    
```

Comment:
Better file error handling just meant more explicit try/catch statements to provide more user-friendly information. This is generally good practice in development instead of letting it fail, but based on what I've heard, that level of error handling isn't necessary here. I can remove the note.

Reply:
Also, not sure why I can't reply to the questions about the column/row skipping:

I added an optional parameter so that users can set the number of rows to skip themselves, but it defaults to 2. I'm pretty sure (based on the files Sam has provided) that the first 2 rows are just extra info and the column headers start on the 3rd row. I'll make sure to check with Sam and confirm that this is still the case.

regarding usecols: I separated validation from reading the files. It will read the file just fine if required columns are missing, but it will fail during validation if it is missing a column that was specified in the schema as belonging to that table.